src/feature/rocket/rocketSlice.js

```diff
@@ -16,7 +16,26 @@ const initialState = {
 const rocketSlice = createSlice({
   name: 'rocket',
   initialState,
-  // reducers: {},
+  reducers: {
+    makeReservation: (state, { payload }) => {
+      const newState = state.rockets.map((rocket) => {
+        if (rocket.id === payload) {
+          return { ...rocket, reserved: true };
+        }
+        return rocket;
+      });
+      state.rockets = newState;
```

Comment:
This is mutating the state.
In the requirement it requests us not to mutate the state
you can use 
`return {..state,rockets: newState}`